src/jemalloc.c

```diff
@@ -1079,6 +1075,24 @@ validate_hpa_settings(void) {
 	    "hpa_hugification_threshold_ratio: %s and hpa_dirty_mult: %s. "
 	    "These values should sum to > 1.0.\n%s", hugification_threshold,
 	    hpa_dirty_mult, normalization_message);
+
+	return true;
+}
+
+static void
+validate_hpa_settings(void) {
+	if (!hpa_supported() || !opt_hpa) {
+		return;
+	}
+	if (HUGEPAGE > HUGEPAGE_MAX_EXPECTED_SIZE) {
+		had_conf_error = true;
+		malloc_printf(
+		    "<jemalloc>: huge page size (%zu) greater than expected.\n",
```

Comment:
Can we also show `HUGEPAGE_MAX_EXPECTED_SIZE` in the error message so that users know how to configure more easily?

Reply:
I don't think we should show the MAX_EXPECTED_SIZE. It's just a way of us guessing that, that setting must not be supported well. Not really meaningful as it can be larger or smaller, and the current value is simply a best guess.

Also note that the huge page size isn't easily configurable. It's tied to hardware TLBs so there's only a handful of supported values. This msg is more saying "it's not supported and don't use it here", instead of "change some setting to a lower value so it could work".

Reply:
Tweaked the wording.